javascript/extractor/src/com/semmle/jcorn/flow/FlowParser.java

```diff
@@ -1203,4 +1207,49 @@ protected boolean atGetterSetterName(PropertyInfo pi) {
 		return super.atGetterSetterName(pi);
 	}
 
+    @Override
+	protected Pair<Expression, Boolean> parseSubscript(final Expression base, Position startLoc, boolean noCalls) {
+		if (!noCalls) {
+			maybeFlowParseTypeParameterInstantiation(base, true);
+		}
+		return super.parseSubscript(base, startLoc, noCalls);
+	}
+
+	private void maybeFlowParseTypeParameterInstantiation(Expression left, boolean requireParenL) {
+		if (flow() && this.isRelational("<")) {
+			// Ambiguous case: `e1<e2>(e3)` is parsed differently as JS and Flow code:
+			// JS: two relational comparisons: `e1 < e2 > e3`
+			// Flow: a call `e1(e3)` with explicit type parameter `e2`
+
+			// Heuristic: if the left operand of the `<` token is a primitive, then it probably isn't a call, as that would always crash
+			left = left.stripParens();
+			if (left instanceof Literal) {
```

Comment:
Are there any literals for which we would _not_ want to choose the JS parse?

Reply:
function literals?

Reply:
Those would be `FunctionExpression`s, no?

Reply:
I see. See the most recent commit: we now use the JS parse for all literals.

Reply:
LGTM, good to proceed with the full evaluation as far as I'm concerned.